packages/jest-core/src/TestScheduler.ts

```diff
@@ -193,16 +207,60 @@ export default class TestScheduler {
     if (testsByRunner) {
       try {
         for (const runner of Object.keys(testRunners)) {
-          await testRunners[runner].runTests(
-            testsByRunner[runner],
-            watcher,
-            onStart,
-            onResult,
-            onFailure,
-            {
-              serial: runInBand || Boolean(testRunners[runner].isSerial),
-            },
-          );
+          const testRunner = testRunners[runner];
+          const context = contextsByTestRunner.get(testRunner);
+          const tests = testsByRunner[runner];
+
+          const testRunnerOptions = {
+            serial: runInBand || Boolean(testRunner.isSerial),
+          };
+
+          /**
+           * Test runners with event emitters are still not supported
+           * for third party test runners.
+           */
+          if (testRunner.__PRIVATE_UNSTABLE_API_supportsEventEmitters__) {
+            const unsubscribes = [
+              testRunner.on('test-file-start', ([test]) =>
+                onTestFileStart(test),
+              ),
+              testRunner.on('test-file-success', ([test, testResult]) =>
+                onResult(test, testResult),
+              ),
+              testRunner.on('test-file-failure', ([test, error]) =>
+                onFailure(test, error),
+              ),
+              testRunner.on(
+                'test-case-result',
+                ([testPath, testCaseResult]) => {
+                  if (context) {
```

Comment:
`context` should always be defined, no? Should we add an `invariant` below the `const context = contextsByTestRunner.get(testRunner);` on line 211?